sound/soc/sof/intel/intel-ipc.c

```diff
@@ -30,10 +30,24 @@ void intel_ipc_msg_data(struct snd_sof_dev *sdev,
 			void *p, size_t sz)
 {
 	if (!substream || !sdev->stream_box.size) {
+		if (sz > sdev->dsp_box.size) {
+			dev_err(sdev->dev, "error: dsp_box read size %ld is out of range %ld",
+				sz, sdev->dsp_box.size);
+
+			return;
+		}
+
 		sof_mailbox_read(sdev, sdev->dsp_box.offset, p, sz);
 	} else {
 		struct intel_stream *stream = substream->runtime->private_data;
 
+		if (sz + stream.posn_offset > sdev->stream_box.size + sdev->stream_box.offset) {
```

Comment:
that should be `stream->posn_offset` above, looks like this hasn't been compile-tested?

Reply:
sorry,  I just realized that I didn't enabled BYT in kconfig.